File: frontend/projects/admin-console/src/app/core/models/evidence-item.dto.ts
export interface EvidenceItemDto {
  id: string;
  entityType: string; // Document, DocumentVersion, AuditAnswer, etc.
  entityId: string; // ID of the parent entity
  fileName: string;
  storageLocation: string;
  fileHash?: string; // SHA256 hex string
  uploadedAt: string; // ISO string
  uploadedBy?: string;
  tenantId?: string;
  
  // Additional context fields
  documentTitle?: string; // For Document/DocumentVersion
  auditRunId?: string; // For AuditAnswer
  auditQuestionId?: string; // For AuditAnswer
  versionNumber?: number; // For DocumentVersion
}

File: application/src/Maemo.Application/Evidence/Dtos/EvidenceItemDto.cs
namespace Maemo.Application.Evidence.Dtos;

/// <summary>
/// DTO representing an evidence item in the Evidence Register.
/// </summary>
public class EvidenceItemDto
{
    public Guid Id { get; set; }
    public string EntityType { get; set; } = null!; // Document, DocumentVersion, AuditAnswer, etc.
    public string EntityId { get; set; } = null!; // ID of the parent entity (DocumentId, AuditRunId, etc.)
    public string FileName { get; set; } = null!;
    public string StorageLocation { get; set; } = null!;
    public string? FileHash { get; set; } // SHA256 hex string
    public DateTime UploadedAt { get; set; }
    public string? UploadedBy { get; set; }
    public Guid? TenantId { get; set; }
    
    // Additional context fields
    public string? DocumentTitle { get; set; } // For Document/DocumentVersion
    public Guid? AuditRunId { get; set; } // For AuditAnswer
    public Guid? AuditQuestionId { get; set; } // For AuditAnswer
    public int? VersionNumber { get; set; } // For DocumentVersion
}

